cmd/snap-repair/runner.go

```diff
@@ -900,8 +906,50 @@ func (run *Runner) Applicable(headers map[string]interface{}) bool {
 		}
 	}
 
-	// TODO:UC20: need to consider filtering by bases and modes in the assertion
-	// here
+	// also filter by base snaps and modes
+	bases, err := stringList(headers, "bases")
+	if err != nil {
+		return false
+	}
+
+	if len(bases) != 0 && !strutil.ListContains(bases, run.state.Device.Base) {
+		return false
+	}
+
+	modes, err := stringList(headers, "modes")
+	if err != nil {
+		return false
+	}
+
+	// modes is slightly more nuanced, if the modes setting in the assertion
+	// header is unset, that means that the assertion should only be run in
+	// "run" mode on uc20
+
+	if run.state.Device.Mode == "" {
+		// uc16 / uc18 device, the assertion is only applicable to us if modes
+		// has "run" in it or if it is unset
+		if len(modes) != 0 && !strutil.ListContains(modes, "run") {
```

Comment:
modes in the assertion can only be non-empty if bases are core2* or empty, so the only relevant case would be a repair with:

```
modes: [run]
(no bases)
```

I think such a repair should also imply core20+ only

Reply:
how do we handle the implication of core20+ here? do we want to hard-code a block list of core and core18, and assume all other base snaps are core20+ like we did previously?

Reply:
my naive thinking was to just have:

```
if len(modes) != 0 {
    return false
}
```

on this path where `Device.Mode == ""`

Reply:
this is done now